File: src/main/scala/robosky/uplands/UplandsBlockTags.scala
package robosky.uplands

import net.fabricmc.fabric.api.tag.TagRegistry
import net.minecraft.block.Block
import net.minecraft.tag.Tag
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry
import net.minecraft.world.biome.Biome

object UplandsBlockTags {
  val AzoteMushroomSpreadable = register("azote_mushroom_spreadable")
  val PlantableOn = register("plantable_on")

  def init(): Unit = {}

  def register(name: String): Tag[Block] = TagRegistry.block(UplandsMod :/ name)
}

File: src/main/java/robosky/uplands/block/UplandsBlockTags.java
package robosky.uplands.block;

import net.fabricmc.fabric.api.tag.TagRegistry;
import robosky.uplands.UplandsMod;

import net.minecraft.block.Block;
import net.minecraft.tag.Tag;

public final class UplandsBlockTags {
	public static final Tag<Block> AZOTE_MUSHROOM_SPREADABLE = register("azote_mushroom_spreadable");
	public static final Tag<Block> PLANTABLE_ON = register("plantable_on");

	public static void init() {}

	private static Tag<Block> register(String name) {
		return TagRegistry.block(UplandsMod.id(name));
	}

	private UplandsBlockTags() {}
}
